.github/workflows/build.yaml

```diff
@@ -0,0 +1,75 @@
+name: build
+on:
+  push:
+  pull_request:
+  schedule:  # nightly build
+    - cron: '0 0 * * *'
+
+jobs:
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - run: python -m pip install tox
+      - run: python -m tox -e mypy
+
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        tox-factor: [""]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        include:
+          - os: windows-latest
+            python-version: "3.10"
+          - os: macos-latest
+            python-version: "3.10"
+          - os: ubuntu-latest
+            python-version: "3.10"
+            tox-factor: "-marshmallowdev"
+          - os: ubuntu-latest
+            python-version: "3.7"
+            tox-factor: "-mindeps"
+    name: "test py${{ matrix.python-version }}${{ matrix.tox-factor }} x ${{ matrix.os }}"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: python -m pip install tox
+      - run: python -m tox -e py${{ matrix.tox-factor }}
+
+  # this duplicates pre-commit.ci, so only run it on tags
```

Comment:
thought (non-blocking): this is probably unnecessary, but i don't feel strongly about removing it.

Reply:
When I added it, my thought was "this is probably unnecessary, but it's easier to add it than to worry about ever having a release tag where linting fails". I don't feel strongly about keeping it. 😉

I probably won't change it today though, since I want to leave this PR untouched for at least a few hours before merging.